authlib/integrations/base_client/async_app.py

```diff
@@ -90,28 +92,60 @@ async def request(self, method, url, token=None, **kwargs):
         async with self._get_oauth_client(**metadata) as session:
             return await _http_request(self, session, method, url, token, kwargs)
 
-    async def create_authorization_url(self, redirect_uri=None, **kwargs):
+    async def create_authorization_url(self, redirect_uri=None, authorize_url_params=None, **authorize_params):
         """Generate the authorization url and state for HTTP redirect.
 
         :param redirect_uri: Callback or redirect URI for authorization.
-        :param kwargs: Extra parameters to include.
+        :param authorize_url_params: Extra parameters specifically for the authorize_url endpoint, regardless of method.
+        :param authorize_params: Extra parameters to include for authorization.
         :return: dict
         """
         metadata = await self.load_server_metadata()
+        if self.authorize_params:
+            authorize_params.update(self.authorize_params)
+
+        authorize_url_params = authorize_url_params or {}
+        if self.authorize_url_params:
+            authorize_url_params.update(self.authorize_url_params)
+
         authorization_endpoint = self.authorize_url or metadata.get(
             "authorization_endpoint"
         )
         if not authorization_endpoint:
             raise RuntimeError('Missing "authorize_url" value')
 
-        if self.authorize_params:
-            kwargs.update(self.authorize_params)
+        par_endpoint = self.pushed_authorization_url or metadata.get(
+            "pushed_authorization_request_endpoint"
+        )
+        par_required = metadata.get("require_pushed_authorization_requests")
+        if par_required and not par_endpoint:
+            raise RuntimeError('Missing "pushed_authorization_url" value')
+        par_required = par_required or par_endpoint
```

Comment:
A detail, I would have called this var `should_use_par` instead.

This makes me think that this enforces the usage of PAR if available, and this is probably a good security practice. However, are there situations when one might want to avoid using PAR even if it is available on the server?